### ISSUE TITLE
New optimizers API accepts 'lr' as argument, but ignores it

### ISSUE THREAD

**Issue by jaspervandorp** [keras-team/keras#17338]
_Note: this was originally posted on TF's Github, but identified as being a Keras-related bug: https://github.com/tensorflow/tensorflow/issues/58831_
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

TF 2.11

### Custom Code

Yes

### OS Platform and Distribution

All

### Python version

3.10.6

### Current Behaviour?

```shell
The 'lr' argument for learning rate got deprecated in recent TF versions, and replaced by 'learning_rate'. Recent TF versions gave a warning when 'lr' was used, which got updated to:

"WARNING:absl:`lr` is deprecated, please use `learning_rate` instead, or use the legacy optimizer, e.g.,tf.keras.optimizers.legacy.Adagrad."

However, training still continuous, ignoring the learning rate set. See the code to reproduce: when 'lr=' is used, the final train loss is 0.1577; when 'learning_rate=' is used the final train loss is 0.5490 (higher loss due to much smaller learning rate).

I think the expected behavior here is that TF throwns an exception when 'lr' is used as an argument. The fact that the optimizer ignores it may easily go unnoticed, despite the warning, e.g. in an MLOps scenario, or when using a third-party module that still uses 'lr=' internally.
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
import tensorflow.keras as keras
from tensorflow.keras.datasets import mnist
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout, Flatten, Activation
from tensorflow.keras.layers import Conv2D, MaxPooling2D
from tensorflow.keras import backend as K
from tensorflow.keras.layers import BatchNormalization
import timeit

print('TF version: {}'.format(tf.__version__))

batch_size = 128
num_classes = 10
epochs = 12
lr = 1e-4

# input image dimensions
img_rows, img_cols = 28, 28

# the data, shuffled and split between train and test sets
(x_train, y_train), (x_test, y_test) = mnist.load_data()

if K.image_data_format() == 'channels_first':
    x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)
    x_test = x_test.reshape(x_test.shape[0], 1, img_rows, img_cols)
    input_shape = (1, img_rows, img_cols)
else:
    x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
    x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
    input_shape = (img_rows, img_cols, 1)

x_train = x_train.astype('float32')
x_test = x_test.astype('float32')
x_train /= 255
x_test /= 255
print('x_train shape:', x_train.shape)
print(x_train.shape[0], 'train samples')
print(x_test.shape[0], 'test samples')

# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)


model = Sequential()
model.add(Conv2D(32, kernel_size=(3, 3),
                 input_shape=input_shape))
model.add(BatchNormalization())
model.add(Activation('relu'))
model.add(Conv2D(64, (3, 3)))
model.add(BatchNormalization())
model.add(Activation('relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))
model.add(Flatten())
model.add(Dense(128))
model.add(BatchNormalization())
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(num_classes, activation='softmax'))

model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adagrad(lr=lr),
              metrics=['accuracy'])

tic = timeit.default_timer()
model.fit(x_train, y_train,
          batch_size=batch_size,
          epochs=epochs,
          verbose=1,
          validation_data=(x_test, y_test))
toc = timeit.default_timer()
print('Total training time: %f sec' % (toc-tic))

score = model.evaluate(x_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])
```

### Relevant log output

_No response_</details>

**Comment by ShisuiUzumaki** [comment]
@tilakraya and @jaspervandorp l I am interested. Could you elaborate more on it? Also, some guidance on solving it would be really helpful

**Comment by jaspervandorp** [comment]
> @tilakraya and @jaspervandorp l I am interested. Could you elaborate more on it? Also, some guidance on solving it would be really helpful

Sure! Are there any details you'd like to know more on? The basic issue is really simple actually: up to recent TF versions, these two were equivalent:

1) AdaGrad(lr=...)
2) AdaGrad(learning_rate=...)

These two only differed in that the first one yielded a warning about 'lr' being deprecated.

In TF2.11, Adagrad() still accepts both keywords, but the learning rate is only set properly in option 2. You can try this out for yourself using the stand-alone example I posted. I would expect that either:

a) The use of 'lr' as a keyword yield an exception, or
b) The use of 'lr' is still equivalent to 'learning_rate' (with a warning)

**Comment by ShisuiUzumaki** [comment]
> > @tilakraya and @jaspervandorp l I am interested. Could you elaborate more on it? Also, some guidance on solving it would be really helpful
> 
> Sure! Are there any details you'd like to know more on? The basic issue is really simple actually: up to recent TF versions, these two were equivalent:
> 
> 1. AdaGrad(lr=...)
> 2. AdaGrad(learning_rate=...)
> 
> These two only differed in that the first one yielded a warning about 'lr' being deprecated.
> 
> In TF2.11, Adagrad() still accepts both keywords, but the learning rate is only set properly in option 2. You can try this out for yourself using the stand-alone example I posted. I would expect that either:
> 
> a) The use of 'lr' as a keyword yields an exception, or b) The use of 'lr' is still equivalent to 'learning_rate' (with a warning)

So, either it should throw an error or work (while giving deprecations warning), right?.

**Comment by jaspervandorp** [comment]
> > > @tilakraya and @jaspervandorp l I am interested. Could you elaborate more on it? Also, some guidance on solving it would be really helpful
> > 
> > 
> > Sure! Are there any details you'd like to know more on? The basic issue is really simple actually: up to recent TF versions, these two were equivalent:
> > 
> > 1. AdaGrad(lr=...)
> > 2. AdaGrad(learning_rate=...)
> > 
> > These two only differed in that the first one yielded a warning about 'lr' being deprecated.
> > In TF2.11, Adagrad() still accepts both keywords, but the learning rate is only set properly in option 2. You can try this out for yourself using the stand-alone example I posted. I would expect that either:
> > a) The use of 'lr' as a keyword yields an exception, or b) The use of 'lr' is still equivalent to 'learning_rate' (with a warning)
> 
> So, either it should throw an error or work (while giving deprecations warning), right?.

Exactly, either one of those two would be expected behavior as far as I'm concerned.

**Comment by tilakrayal** [comment]
@jaspervandorp,
I can see that it is providing the warning `UserWarning: The `lr` argument is deprecated, use `learning_rate` instead.` Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/4c62c6705e67be925b93c9e6adfba12b/untitled806.ipynb).

**Comment by jaspervandorp** [comment]
> @jaspervandorp, I can see that it is providing the warning `UserWarning: The `lr`argument is deprecated, use`learning_rate` instead.` Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/4c62c6705e67be925b93c9e6adfba12b/untitled806.ipynb).

Correct, and this warning has been there for a while now, with one major difference: in previous versions 'lr=' still worked as a keyword, despite the warning. I think that now 'lr=' no longer sets the learning rate appropriately, the warning should be an exception. As explained in the original report, the warning can be easily missed in a scenario where someone is running their ML within a MLOps environment, or using third-party scripts, for example.

But maybe I'm wrong in thinking this?

**Comment by ShisuiUzumaki** [comment]
So, should we throw an exception or not?

**Comment by jaspervandorp** [comment]
> If it has been there for a while, like for a few versions, maybe it is a good idea to start removing it with the initial steps of exceptions since it is intended to be removed eventually. It is just a personal thought. What do you think @tilakrayal ?

I would agree with this. Either remove it (with an exception) or keep it in place and working (with a warning)

**Comment by haifeng-jin** [comment]
@chenmoneygithub Would you please comment on this issue? Thanks.

**Comment by chenmoneygithub** [comment]
@jaspervandorp What actually happens is `lr` is silently ignored, agree we should throw an exception. Will work on that.

**Comment by ShisuiUzumaki** [comment]
> @jaspervandorp What actually happens is `lr` is silently ignored, agree we should throw an exception. Will work on that.

@chenmoneygithub Can I work on it as a contributor?
